test/server.js

```diff
@@ -465,50 +465,44 @@ describe("server", function() {
                 }).then(function(response) {
                     var afterTimestamp = new Date();
                     assert.equal(helpers.getUpdateOneCallCount("notifications"), 2);
+                    // Verify that server has generated valid timestamp values
                     var firstTimestamp = helpers.getUpdateOneArgs("notifications", 0)[1].$set["data.since"];
                     var secondTimestamp = helpers.getUpdateOneArgs("notifications", 1)[1].$set["data.since"];
                     assert.isAtLeast(firstTimestamp.getTime(), beforeTimestamp.getTime());
                     assert.isAtMost(firstTimestamp.getTime(), afterTimestamp.getTime());
                     assert.equal(firstTimestamp, secondTimestamp);
-                    assert.deepEqual(helpers.getUpdateOneArgs("notifications", 0), [
-                        {
+                    // Verify that notification upserts have been performed with correct values
```

Comment:
The function is still quite long, largely because of the size of the objects being tested, but I've tried to improve the readability of the function.